### ISSUE TITLE
ColumnReader.Skip can give access violation for Nullable<bool> field

### ISSUE THREAD

**Issue by TobyShaw** [G-Research/ParquetSharp#229]
Code repro: 

```fsharp
open ParquetSharp
open ParquetSharp.IO

let resizableBuffer = new ResizableBuffer()

let column = Column<bool Nullable> "Value"
do
    // Set-up: Make a file with one row-group, one column, the column is filled with 100,000 nullable bools.
    // Note about F# syntax: once we exit the do-block, all writers will have been disposed in the correct order.
    use outputStream = new BufferOutputStream(resizableBuffer)
    use fileWriter = new ParquetFileWriter(outputStream, [|column :> Column|])
    use rowGroup = fileWriter.AppendRowGroup()
    use logicalWriter = rowGroup.NextColumn().LogicalWriter<bool Nullable>()
    logicalWriter.WriteBatch [|for i in 1..100_000 -> Nullable (i % 2 = 0)|]

// Get a column reader for this Nullable<bool> column
use fr = new ParquetFileReader(new BufferReader(resizableBuffer))
use rgr = fr.RowGroup 0
let columnReader = rgr.Column(0)

// When we skip to halfway through the column, we get an access violation very consistently.
columnReader.Skip(50_000L) |> printfn "We crash before reaching this: %A"
```
I can reproduce this error with a file on disk, as well as an in-memory buffer (it just happens to be easier to give a repro using the in-memory buffer)

This was observed on versions 2.2.0 and 5.0.0-beta1 of ParquetSharp

**Comment by jgiannuzzi** [comment]
Thanks for your bug report, @TobyShaw!

We will look into this.

**Comment by adamreeve** [comment]
Hi @TobyShaw, this is a bug in the underlying Apache Arrow library (see https://issues.apache.org/jira/browse/ARROW-11518). This has been fixed in Arrow 6.0.0, so should be fixed by #230.

**Comment by jgiannuzzi** [comment]
Fixed in [ParquetSharp 6.0.1-beta1](https://github.com/G-Research/ParquetSharp/releases/tag/6.0.1-beta1)
